src/v/redpanda/debug_bundle.cc

```diff
@@ -0,0 +1,292 @@
+                                      "undefined");
+                                }
+
+                                if (exit_status->exit_code != 0) {
+                                    vlog(
+                                      bundle_log.error,
+                                      "Failed to run RPK, exit code {}",
+                                      exit_status->exit_code);
+                                } else {
+                                    vlog(
+                                      bundle_log.debug,
+                                      "RPK successfully created debug bundle "
+                                      "{}",
+                                      _in_progress_filename);
+                                }
+                            })
+                            .finally([&p, guard{std::move(guard)}]() mutable {
+                                // Make sure the process dies, first gracefully
+                                // and then forcefully.
+                                // Please note: Seastar reports an ignored
+                                // exceptional future when SIGTERM or SIGKILL is
+                                // called on an already dead process.
+                                p.terminate();
+                                p.kill();
+                            });
+                      });
+                });
+          });
+
+    return ss::make_ready_future<>();
+}
+
+ss::future<debug_bundle_status> debug_bundle::get_status() {
+    if (ss::this_shard_id() != debug_bundle_shard_id) {
+        co_return co_await container().invoke_on(
+          debug_bundle_shard_id,
+          [](debug_bundle& b) mutable { return b.get_status(); });
+    }
+
+    co_return _status;
+}
+
+ss::sstring detail::make_bundle_filename(
+  const std::filesystem::path& write_dir, ss::sstring& filename) {
+    return fmt::format("{}/{}", write_dir.string(), filename);
+}
+
+void detail::throw_if_bundle_dne(
+  debug_bundle_status bundle_status,
+  const std::filesystem::path& write_dir,
+  ss::sstring& filename) {
+    auto bundle_name = detail::make_bundle_filename(write_dir, filename);
+    if (!std::filesystem::exists(bundle_name.c_str())) {
+        if (bundle_status == debug_bundle_status::running) {
+            throw ss::httpd::not_found_exception(
+              "The debug bundle is running but not yet ready");
+        } else {
+            throw ss::httpd::base_exception(
+              "The debug bundle is not running or it is not on disk",
+              ss::http::reply::status_type::gone);
+        }
+    } else {
+        if (bundle_status == debug_bundle_status::running) {
+            throw ss::httpd::not_found_exception(
+              "The debug bundle is running but not yet ready");
+        }
+    }
+}
+
+ss::future<std::unique_ptr<ss::http::reply>> debug_bundle::fetch_bundle(
+  std::unique_ptr<ss::http::request> req,
+  std::unique_ptr<ss::http::reply> rep) {
```

Comment:
So this one is a little trickier to de-couple from the HTTP service.  Maybe it should return the path to the file so the HTTP service can handle the file transfer.

Reply:
>So this one is a little trickier to de-couple from the HTTP service. Maybe it should return the path to the file so the HTTP service can handle the file transfer.

Two things here. First, yes, we should get rid of the http bits here. Michaels idea of returning the file path makes a lot of sense to me.

The other general issue in here is that: it seems like it is problematic to move the req/resp across cores like is happening in this PR? I don't know if that is true in this case, it's just that it's true for most things in seastar.

Reply:
This makes sense. I will remove the httpd subsystem from this class.